pkg/components/internal/testutil/jsonpath.go

```diff
@@ -15,70 +15,65 @@
 package testutil
 
 import (
+	"fmt"
 	"reflect"
+	"strings"
 	"testing"
 
+	"github.com/kinvolk/lokomotive/pkg/k8sutil"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	yamlserializer "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
 	"k8s.io/client-go/util/jsonpath"
 )
 
-// unstructredObj accepts a Kubernetes manifest in YAML format and returns an object of type
-// `unstructured.Unstructured`. This object has many methods that can be used by the consumer to
-// extract metadata from the Kubernetes manifest.
-func unstructredObj(t *testing.T, yamlObj string) *unstructured.Unstructured {
-	u := &unstructured.Unstructured{}
-
-	// Decode YAML into `unstructured.Unstructured`.
-	dec := yamlserializer.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
-	if _, _, err := dec.Decode([]byte(yamlObj), nil, u); err != nil {
-		t.Fatalf("Converting config to unstructured.Unstructured: %v", err)
-	}
-
-	return u
-}
-
 // valFromObject takes a JSON path as a string and an object of type `unstructured.Unstructured`.
 // This function returns an object of type `reflect.Value` at that JSON path.
-func valFromObject(t *testing.T, jp string, obj *unstructured.Unstructured) reflect.Value {
+func valFromObject(jp string, obj *unstructured.Unstructured) (reflect.Value, error) {
 	jPath := jsonpath.New("parse")
 	if err := jPath.Parse(jp); err != nil {
-		t.Fatalf("Parsing JSONPath: %v", err)
+		return reflect.Value{}, fmt.Errorf("parsing JSONPath: %w", err)
 	}
 
 	v, err := jPath.FindResults(obj.Object)
 	if err != nil {
-		t.Fatalf("Finding results using JSONPath in the YAML file: %v", err)
+		return reflect.Value{}, fmt.Errorf("finding results using JSONPath in the YAML file: %w", err)
 	}
 
 	if len(v) == 0 || len(v[0]) == 0 {
-		t.Fatalf("No result found")
+		return reflect.Value{}, fmt.Errorf("no result found")
 	}
 
-	return v[0][0]
+	return v[0][0], nil
 }
 
 // jsonPathValue extracts an object at a JSON path from a YAML config, and returns an interface
 // object.
-func jsonPathValue(t *testing.T, yamlConfig string, jsonPath string) interface{} {
-	u := unstructredObj(t, yamlConfig)
-	got := valFromObject(t, jsonPath, u)
+func jsonPathValue(yamlConfig string, jsonPath string) (interface{}, error) {
```

Comment:
Ditto commit.

Reply:
Ah, I see this is used in other commit. Perhaps this change should be moved to where you add `JSONPathExists` (which maybe should also be done in a separate commit), separate from actually using this function.